lib/matplotlib/backend_tools.py

```diff
@@ -99,20 +98,6 @@ def trigger(self, sender, event, data=None):
 
         pass
 
-    @figure.setter
-    def figure(self, figure):
-        """
-        Set the figure
-
-        Set the figure to be affected by this tool
-
-        Parameters
-        ----------
-        figure: `Figure`
-        """
-
-        self._figure = figure
```

Comment:
What about multifigure manager?  It would make sense to me to change the toolmanager to relate to a different figure.  Especially if they use a separate window for a toolbar like in your ``gui_elements`` example.